src/components/fx/hover.js

```diff
@@ -700,9 +713,26 @@ function _hover(gd, evt, subplot, noHoverEvent, eventTarget) {
     gd._spikepoints = newspikepoints;
 
     var sortHoverData = function() {
-        if(hoversubplots !== 'axis') {
-            hoverData.sort(function(d1, d2) { return d1.distance - d2.distance; });
-        }
+        // When sorting keep the points in the main subplot at the top
+        // then add points in other subplots
+
+        var hoverDataInSubplot = hoverData.filter(function(a) {
```

Comment:
naming suggestion -- should these functions be called `hoverDataInMainSubplot` and `hoverDataNotInMainSubplot` then?